app/src/main/java/io/gnosis/safe/ui/transactions/details/view/TxSettingsActionView.kt

```diff
@@ -0,0 +1,120 @@
+package io.gnosis.safe.ui.transactions.details.view
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.Gravity
+import android.widget.LinearLayout
+import io.gnosis.data.models.TransactionInfo
+import io.gnosis.data.repositories.SafeRepository.Companion.DEFAULT_FALLBACK_HANDLER
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_ADD_OWNER_WITH_THRESHOLD
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_CHANGE_MASTER_COPY
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_CHANGE_THRESHOLD
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_DISABLE_MODULE
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_ENABLE_MODULE
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_REMOVE_OWNER
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_SET_FALLBACK_HANDLER
+import io.gnosis.data.repositories.SafeRepository.Companion.METHOD_SWAP_OWNER
+import io.gnosis.data.repositories.getValueByName
+import io.gnosis.safe.R
+import io.gnosis.safe.ui.settings.view.AddressItem
+import io.gnosis.safe.ui.settings.view.LabeledAddressItem
+import io.gnosis.safe.ui.transactions.getVersionForAddress
+import io.gnosis.safe.utils.dpToPx
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+
+class TxSettingsActionView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+    init {
+        orientation = LinearLayout.VERTICAL
+        gravity = Gravity.LEFT
+    }
+
+    private fun clear() {
+        removeAllViews()
+    }
+
+    fun setActionInfo(txInfo: TransactionInfo) {
+        clear()
+        val settingsChange = txInfo as TransactionInfo.SettingsChange
+        settingsMethodTitle[settingsChange.dataDecoded.method]?.let { addStringItem(it) }
+
+        val params = settingsChange.dataDecoded.parameters
+        when (settingsChange.dataDecoded.method) {
+            METHOD_CHANGE_MASTER_COPY -> {
+                val mainCopy = params.getValueByName("_masterCopy")?.asEthereumAddress()
```

Comment:
as mentioned view should be dumb, get everything it needs, and not have this complex parsing logic